File: Python/SortingAlgorithms/insert_sort.py
from typing import TypeVar, List

T = TypeVar("T")

def insert_sort(arr: List[T]) -> None:
    n = len(arr)
    for i in range(1, n):
        current = arr[i]
        j = i - 1
        while j >= 0 and arr[j] > current:
            arr[j+1] = arr[j]
            j = j - 1
        arr[j+1] = current

File: C++/SortingAlgorithms/InsertSort.cpp
#include "InsertSort.h"

template <typename T>
void insert_sort(std::vector<T>& arr)
{
	int size = arr.size();

	for (int i = 1; i < size; i++) {

		T current = arr[i];
		int j = i - 1;

		while (j >= 0 && arr[j] > current) {

			arr[j + 1] = arr[j];
			j--;
		}

		arr[j + 1] = current;
	}
}


template void insert_sort<int>(std::vector<int>&);